packages/celery-library/tests/unit/test_async_jobs.py

```diff
@@ -301,6 +301,8 @@ async def test_async_jobs_cancel(
         payload=60 * 10,  # test hangs if not cancelled properly
     )
 
+    await asyncio.sleep(3.0)    # wait a bit before cancelling
```

Comment:
we have usually very bad experience in terms of flakyness with arbitrary sleeps. Is this to ensure context switching? (in which case `await asyncio.sleep(0)` is enough), otherwise can't we use tenacity?

Reply:
Yup, here we must ensure that task has been picked up by the worker. Let's see if I can find something else.